test/EndToEndSpec.hs

```diff
@@ -555,6 +729,53 @@ spec = around_ (inTempDirectoryNamed "foo") $ do
             other-modules:
                 Baz
           |]
+
+        it "accepts generated-other-modules and infers other-modules" $ do
+          touch "src/Main.hs"
+          touch "src/Foo.hs"
+          [i|
+          executables:
+            foo:
+              main: Main.hs
+              source-dirs: src
+              generated-other-modules: XYZ
+          |] `shouldRenderTo` Package "foo" "Simple" ">= 2.0" [i|
+            executable foo
+              main-is: Main.hs
+              hs-source-dirs:
+                  src
+              other-modules:
+                  Foo
+                  Paths_foo
+                  XYZ
+              autogen-modules:
+                  XYZ
+              default-language: Haskell2010
+          |]
```

Comment:
I think this can be
```
|] `shouldRenderTo` executable "foo" ([i|
main-is: Main.hs
hs-source-dirs:
    src
other-modules:
    Foo
    Paths_foo
    XYZ
autogen-modules:
    XYZ
|]) {packageCabalVersion = ">= 2.0"}
```
instead.